app/views/forms/_odk_group.xml.erb

```diff
@@ -0,0 +1,20 @@
+<group appearance="field-list">
+  <label><%= qing.first.parent.group_name %></label>
+  <% grid_mode = grid_mode?(qing) %>
+
+  <% if grid_mode %>
+    <%= render('odk_questioning',
+      qing: qing.first,
+      style: style,
+      grid_mode: grid_mode,
+      options: {appearance: 'label'}) %>
+  <% end %>
+
+  <% qing.each do |q| %>
+    <%= render('odk_questioning',
+      qing: q,
+      style: @style,
+      grid_mode: grid_mode,
+      options: {appearance: appearance(grid_mode)}) unless q.hidden %>
```

Comment:
Same here. We already have grid_mode inside the partial, and `label_row` will be false by default. So then you can figure out whether/what appearance attrib inside the partial. then you can also nuke the `appearance` method.